dash/quorum/validator_conn_executor.go

```diff
@@ -74,7 +81,9 @@ func NewValidatorConnExecutor(
 	nodeID p2p.ID,
 	eventBus *types.EventBus,
 	sw Switch,
-	logger log.Logger) *ValidatorConnExecutor {
+	logger log.Logger,
+	opts ...optionFunc,
```

Comment:
This is overcomplicating things.
Maybe just add `initialValidatorSet` arg?

Reply:
I do not think so, it is a common approach how to use optional arguments at function/method.
the idea is to avoid a function signature with more than 4 arguments.
btw, I also made `logger` as an optional argument to make the function arguments is a more neat